### PR TITLE
Better source gen debug

### PR DESCRIPTION
**Pull request by TechnologicalPizza**
# Summary of the PR
Moved old debugging code and put it under a compilation conditional.

# Related issues, Discord discussions, or proposals
<https://discord.com/channels/521092042781229087/935115425732263966/935156111361859595>

# Further Comments
Solution files are horrible 🙃

### CODE REVIEW

File: src/Core/Silk.NET.SilkTouch/NativeApiGenerator.cs [hunk 1]
```diff
@@ -157,11 +166,11 @@ INamedTypeSymbol sharedClassSymbol
             
             bool compactFileFormat;
 
-            #if DEBUG
+#if DEBUG
```
Comment (Perksey) [inline src/Core/Silk.NET.SilkTouch/NativeApiGenerator.cs:169]: Does this branch also get hit when `DEBUGANALYZER` is on? i.e. is `DEBUG` a superclass of `DEBUGANALYZER`?

**Comment by Perksey** [comment]
> Solution files are horrible 🙃

yes

File: build/nuke/Silk.NET.NUKE.csproj [hunk 2]
```diff
@@ -8,6 +8,7 @@
     <NukeRootDirectory>..\..</NukeRootDirectory>
     <NukeScriptDirectory>..\..</NukeScriptDirectory>
     <LangVersion>preview</LangVersion>
+    <Configurations>Debug;Release;DebugAnalyzer</Configurations>
```
[the remark this answers is not in the document]
Comment (HurricanKai) [reply build/nuke/Silk.NET.NUKE.csproj:11]: probably bindings.props - that's where SilkTouch is referenced iirc

[the remark this answers is not in the document]
Comment (Perksey) [reply build/nuke/Silk.NET.NUKE.csproj:11]: ah yes that one

[the remark this answers is not in the document]
Comment (Perksey) [reply build/nuke/Silk.NET.NUKE.csproj:11]: Unless we wanted this configuration in every project? As I noticed it's also here in the NUKE project (and other manual ones)

[the remark this answers is not in the document]
Comment (TechnologicalPizza) [reply build/nuke/Silk.NET.NUKE.csproj:11]: Pretty much every project that uses SilkTouch would need this.

[the remark this answers is not in the document]
Comment (Perksey) [reply build/nuke/Silk.NET.NUKE.csproj:11]: Yep, bindings.props then :)

[the remark this answers is not in the document]
Comment (TechnologicalPizza) [reply build/nuke/Silk.NET.NUKE.csproj:11]: I'll remove `<Configurations>` from files that do not use SilkTouch then.

[the remark this answers is not in the document]
Comment (Perksey) [reply build/nuke/Silk.NET.NUKE.csproj:11]: What's stopping you from putting `Configurations` in bindings.props?

[the remark this answers is not in the document]
Comment (TechnologicalPizza) [reply build/nuke/Silk.NET.NUKE.csproj:11]: I'm putting it in binding.props and removing it everywhere else.

[the remark this answers is not in the document]
Comment (Perksey) [reply build/nuke/Silk.NET.NUKE.csproj:11]: Awesoem, thanks.

**Comment by TechnologicalPizza** [comment]
Horrible news. Seems like `<Configurations>` inside `bindings.props` does not work. I get a harsh error upon changing configuration to `DebugAnalyzer`. 
image
Visual Studio seems to show plenty of weird error marks in Solution Explorer as well.
image

And as a last nail in the coffin, selecting `DebugAnalyzer` does not work for `#if DEBUG` conditionals.

**Comment by TechnologicalPizza** [comment]
And as an added bonus, it looks like VS shredded the file endings and encodings on all project files 😑

**Comment by Perksey** [comment]
Hrmf, I can deal with having an extra line in bindings (just please make sure you tell BuildTools about that: it's in a file called ProjectWriter.cs) if you think it's worth it @HurricanKai?

**Comment by HurricanKai** [comment]
I think we should be fine just having the infra (so the changes to the generator itself) but not the configuration - for debugging I generally create a separate project anyways and run it there, since debugging something like Vulkan is basically impossible anyways

**Comment by TechnologicalPizza** [comment]
I'm personally not too happy with:
 1. VS choking on this.
 2. the amount of git diff.
 
This build configuration seems like a bad way forward. Maybe a lockfile was the more user friendly option😢

**Comment by Perksey** [comment]
Closing due to inactivity.

**Pull request closed without merging**